renderer/item_main.js

```diff
@@ -234,6 +289,172 @@ function openExcel() {
     });
 }
 
+//BATCH UPLOAD FUNCTIONS
+/**
+ * Reads a table and generates items from it
+ *
+ * @returns an array of items
+ */
+function getItemsFromTable(tableId) {
+    let table=document.getElementById(`${tableId}`);
+    //find Description, UOM, Commodity Group, and GL Class
+    let rows = parseInt(table.getAttribute("data-rows"));
+    let cols = parseInt(table.getAttribute("data-cols"));
+    //iniitalize items array
+    let items = [];
+    //go through first row to find headings.
+    let validParams = 0;
+    for(let i = 1; i<=cols; i++){
+        //get a cell in the table by its id
+        let cell = document.getElementById("1-"+i);
+
+        //see if cell value matches any of the required parameters to create an item object
+        if(cell.innerHTML.toUpperCase()==='DESCRIPTION'){
+            colLoc.description=i;
+            validParams++;
+        } else if(cell.innerHTML.toUpperCase()==='UOM'||cell.innerHTML.toUpperCase()==='ISSUE UNIT'){
+            colLoc.uom=i;
+            validParams++;
+        } else if(cell.innerHTML.toUpperCase()==='COMMODITY GROUP' || cell.innerHTML.toUpperCase()==='COMM GROUP'){
+            colLoc.commGroup=i;
+            validParams++;
+        } else if(cell.innerHTML.toUpperCase()==='GL CLASS'){
+            colLoc.glClass=i;
+            validParams++;
+        } else if(cell.innerHTML.toUpperCase()==='MAXIMO' || cell.innerHTML.toUpperCase()==='ITEM NUMBER'){
+            colLoc.maximo=i;
+            validParams++;
+        } else if(cell.innerHTML.toUpperCase()==='SERIES'){
```

Comment:
would it be easier to just say the columns must be in a specific order?

Reply:
ya we can say it in the docs, it was just easier to code it this way